prime-router/src/test/kotlin/serializers/Hl7SerializerTests.kt

```diff
@@ -635,6 +636,16 @@ NTE|1|L|This is a final comment|RE"""
         }
     }
 
+    @Test
+    fun `test setTruncationLimitWithEncoding`() {
+
+        val testValue = "Test & Value ~ Text ^ String"
```

Comment:
When doing unit tests, you need to cover all scenarios.  In this case you are missing the no special characters scenario to cover the else in the if statement.

Reply:
got this fixed